RecoParticleFlow/PFClusterProducer/plugins/ECAL2DPositionCalcWithDepthCorr.cc

```diff
@@ -140,7 +140,7 @@ calculateAndSetPositionActual(reco::PFCluster& cluster) const {
     if( rh_energy > 0.0 ) weight = std::max(0.0,( _param_W0 + 
 						  vdt::fast_log(rh_energy) + 
 						  logETot_inv ));
-    const std::shared_ptr<CaloCellGeometry> cell = ((CaloSubdetectorGeometry*)(ecal_geom))->getGeometry(refhit->detId());
+    std::shared_ptr<const CaloCellGeometry> cell = ((CaloSubdetectorGeometry*)(ecal_geom))->getGeometry(refhit->detId());
```

Comment:
You should not be doing a C style cast here (you should not be doing a cast in any of the code you changed).